src/pages/todo/components/TodoList.js

```diff
@@ -2,29 +2,105 @@ import React from 'react';
 import { provideHooks, trigger } from 'redial';
 import { connect } from 'react-redux';
 import { Link } from 'react-router'
-import { LIST_TODOS } from '../actions';
+import { LIST_TODOS, CREATE_TODO } from '../actions';
+import TodoListItem from './TodoListItem';
 
 @provideHooks({
   fetch: ({ dispatch, params: { id } }) => dispatch({type: LIST_TODOS.REQUEST})
 })
 @connect((state) => {
   return {
-    todos: state.todoReducer && state.todoReducer.todos || []
+    todos: state.todoReducer && state.todoReducer.todos || [],
```

Comment:
Please add fetchState to  todoReducer as follows
```
CREATE_TODO: state.todoReducer || null
```
to
```
fetchState: state.todoReducer || state.todoReducer.fetchState || {}
```

```
if(nextProps[CREATE_TODO.BASE]) {
    const {error, fetching} = nextProps[CREATE_TODO.BASE];
```
to
```
if(nextProps.fetchState[CREATE_TODO.BASE]) {
      const {error, fetching} = nextProps.fetchState[CREATE_TODO.BASE];
```

```
nextState = Object.assign({}, state);
     nextState[requestTypes.BASE] = {error: null, fetching: true};
     return nextState;
```
to
```
let fetchState = {};
fetchState[requestTypes.BASE] = {error: null, fetching: true};
return Object.assign({}, state, fetchState);
```
(same for others)